Week1 [STEP 3] 아담

yawoong2 opened the pull request:
@Tediousday93 

안녕하세요 로웬! 늦은 밤입니다. Step 3에 대한 PR이에요!

 ## 고민했던 점
 ### 빼빼로 몸통 모양에 따른 막대 길이 조절
 - 빼빼로 몸통 형식에 따라 (ex: "***", "&***&", " |0|" 등) 막대의 좌우 정렬을 맞추려고 시도해보았습니다.
 - 제가 제출한 코드에서는 4가지 제시된 입력 예시에만 적합하도록 삼항연산자를 사용했습니다.
 ---

 ## 해결이 되지 않은 점
 ### 빼빼로 몸통 모양에 따른 막대 길이 조절
 - 그러나 제출한 코드는 4가지의 입력 예시에만 알맞은 코드일뿐, 만약 빼뺴로 몸통이 (ex: *******) 처럼 너비가 넓어진다면, 막대가 가운데 정렬이 되지 않습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 막대 좌측의 공백의 갯수를 결정하는 법
 - 제출한 코드 이외에 막대를 그리는 함수 내 이중 for문을 사용해서 "| |" 좌측에 띄어쓰기를 삽입하여 위 문제를 해결하긴 했으나, 이중 for문을 사용하지 않고 막대의 가운데 정렬을 할 수 있는 방법이 있을까요?
 ```

yawoong2 committed:
feat: Step 3 빼빼로만들기 기능 추가

Step3 빼빼로만들기 기능 추가 및 스텝별 swift파일 분할

yawoong2 committed:
docs: 빼빼로 만들기 순서도 작성

step 3: 빼빼로 만들기 함수에 대한 순서도 작성

Tediousday93 reviewed (changes requested):
안녕하세요 아담님,
이번 스텝도 정말 잘 수행해주셨습니다!!
코멘트 드릴 것이 거의 없네요 😅

* naming convention - camelCase 완벽히 지켜주셨습니다 👍
* main.swift 파일과 다른 파일의 차이점에 대해 알고 계신 것 같습니다 👍

고생하셨습니다!

아래 코드에서도 코멘트를 확인해주시고, 수정이 완료되시면 디스코드로 머지 의사를 알려주세요 ☺️

---

### 수행한 점

* [x] 정확한 브랜치에 PR 보내기
* [x] PR의 제목/내용을 양식에 맞게 보내기
* [x] 학습 도우미 태그하기

---

* [x] Step2와 같은 프로젝트에서 새로운 파일을 작성하기
* [x] 요구사항에 맞는 함수 만들기
* [x] 다양한 빼빼로 출력하기
* [x] 순서도 만들기

---

</br>

### ✅ 고민했던 점

**빼빼로 몸통 모양에 따른 막대 길이 조절**
* 사실 빼빼로 몸통의 가로 문자 개수가 짝수가 되면 막대가 굵기가 변경되거나 중앙 정렬이 어렵죠 😅
* 삼항연산자를 활용하여 주어진 조건을 해결하신 부분 👍
* 확장성을 고려하여 리팩토링을 해볼 수 있겠습니다.

</br>

### ✅ 해결이 되지 않은 점

**막대 길이 조절**
확장성을 이미 고려하고 계시고 있군요👍 해결하는 방법은 다양할 것 같습니다.
* 파라미터에 직접 body에 맞는 stick 문자열을 받는 방법
* 삼항연산자(또는 조건문)를 활용하기 위해 body의 중앙 문자 count를 구하여 그만큼 왼쪽에 공백을 주는 방법
* 등등...

</br>

### ✅ 조언을 얻고 싶은 부분

**막대 좌측 공백 개수 구하기**
* 이중 for문은 지양하는 것이 좋습니다! 이중 for 문을 통해 해결하셨다면 안쪽의 for-in 반복문을 별도의 함수로 분리해보시면 좋을 거에요 🤔
* tip: `공백을 구하여 문자열에 추가하는 기능`은 막대를 그리는 기능과는 다른 기능이므로 함수를 통해 기능을 분리해준다고 생각하시면 좋습니다!
* 질문이 생기시면 언제든 DM 보내주세요 ☺️ (질문하실 때는 캠핑 사이트 필독사항의 `⁉️ 질문과 답변, 그리고 공유하기`를 참고하면 서로에게 도움이 된답니다!)

yawoong2 committed:
refactor: 빼빼로 막대 그리기 기능을 분리

- 빼빼로 몸통 너비에 따라, 빼빼로 막대를 가운데 정렬하기 위하여
막대 그리기 기능(함수)를 분리함.
- 이해를 높일 수 있도록 변수 및 함수명 변경

yawoong2 committed:
refactor: drawWholeStick 함수 수정

drawWholeStick 함수 내부에 drawStickSample 함수를 호출할 수 있도록 수정

yawoong2 committed:
docs: 빼빼로 만들기 순서도 수정

- 반복문을 표현하는 다른 도형을 이용하여 순서도 표현
- 코드 로직 변경에 따른 순서도 내용 수정
